### PR TITLE
Update readme with feature matrix

### PR DESCRIPTION
**Pull request by yzhong94**
Updated feature matrix for device and service SDK.  @prmathur-microsoft  Can you look over the readme?  Some sections need to be vetted by you.

This should go live on Monday morning with the blog

# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,32 +1,86 @@
...
-* **Azure IoT Hub Device SDK**: to connect client devices to Azure IoT Hub (supports Java 7+ and Android API 17+)
-* **Azure IoT Hub Service SDK**: enables developing back-end applications for Azure IoT (supports Java 7+)
+* **Azure IoT Hub device SDK for Java**: to connect client devices to Azure IoT Hub (supports Java 7+ and Android API 17+)
+* **Azure IoT Hub service SDK for Java**: enables developing back-end applications for Azure IoT (supports Java 7+)
+
+The API reference documentation for the device SDK is [here][java-api-reference-device].
+
+The API reference documentation for the service SDK is [here][java-api-reference-service].
 
 To find SDKs in other languages for Azure IoT, please refer o the [azure-iot-sdks][azure-iot-sdks] repository
```
Comment (prmathur-microsoft) [inline readme.md]: While you are at it please correct typo - refer to

File: readme.md [hunk 2]
```diff
@@ -1,32 +1,86 @@
...
-The API reference documentation for the device SDK is [here][java-api-reference-device].
-The API reference documentation for the service SDK is [here][java-api-reference-service].
 * On Linux and Windows:
    * **Using Maven**: the simplest way to use the Azure IoT SDKs for Java to develop apps is to leverage Maven packages:
       * [Device SDK][device-maven]
       * [Service SDK][service-maven]
+   * **Clone the repository**: The repository is using [GitHub Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) for its dependencies. In order to automatically clone these submodules, you need to use the --recursive option as described here:
+   ```
+    git clone --recursive https://github.com/Azure/azure-iot-sdk-java.git  
+   ```
```
Comment (prmathur-microsoft) [inline readme.md]: Java does not use submodule and we should not be. Why was this needed ?

File: readme.md [hunk 3]
```diff
@@ -1,32 +1,86 @@
...
+|------------------------------------------------------------------------------------------------------------------|--------------------------|--------------------------|--------------------------|--------------------------|--------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [Authentication](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-security-deployment)                     | :heavy_check_mark:       | :heavy_check_mark:*      | :heavy_check_mark:*      | :heavy_check_mark:*      | :heavy_check_mark:       | Connect your device to IoT Hub securely with supported authentication, including private key, SASToken, X-509 Self Signed and Certificate Authority (CA) Signed over MQTT.  *IoT Hub only supports X-509 CA Signed over AMQP and MQTT at the moment.  X509-CA Signed over AMQP is in progress.                                              |
```
Comment (prmathur-microsoft) [inline readme.md]: Incorrect about AMQP. I will send you email with details on X509 support. Please update this accordingly.

File: readme.md [hunk 4]
```diff
@@ -1,32 +1,86 @@
...
+| [Receive cloud-to-device messages](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-c2d) | :heavy_check_mark:*      | :heavy_check_mark:*      | :heavy_check_mark:*      | :heavy_check_mark:*      | :heavy_check_mark:       | Receive cloud-to-device messages and read associated custom and system properties from IoT Hub, with the option to complete/reject/abandon C2D messages.  *C2D message size limit of 256 KB over MQTT and AMQP is in progress.  IoT Hub supports the option to complete/reject/abandon C2D messages over HTTPS and AMQP only at the moment. |
+| [Device Twins](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-device-twins)                     | :heavy_multiplication_x: | :heavy_multiplication_x: | :heavy_multiplication_x: | :heavy_multiplication_x: | :heavy_minus_sign:       | IoT Hub persists a device twin for each device that you connect to IoT Hub.  The device can perform operations like get twin tags, subscribe to desired properties.  *Send reported properties version and desired properties version are in progress.                                                                                      |
```
Comment (prmathur-microsoft) [inline readme.md]: This is bad way of putting how certain API's dont exist on device twin. Reading this, users will see it as Device twin is not supported which is incorrect statement

**Comment by prmathur-microsoft** [comment]
Also please note we don't use multiple commit messages. Please specify one commit message that describes this change and get rid of all other descriptions.

File: readme.md (hunk 1 above)
Comment (yzhong94) [reply readme.md]: Corrected

File: readme.md (hunk 2 above)
Comment (yzhong94) [reply readme.md]: Updated

File: readme.md (hunk 4 above)
Comment (yzhong94) [reply readme.md]: You are correct, wrong notation.  Should have been check*

File: readme.md (hunk 3 above)
Comment (yzhong94) [reply readme.md]: Updated.  X509 over AMQP is not supported right?

**Comment by prmathur-microsoft** [comment]
This PR is merged. Thanks for contributing.
